File: modules/shared/src/main/scala/shared/topics/TopicTypes.scala
package shared.topics

object TopicTypes {
  object CommandTopic {
    val request  = "command_request"
    val response = "command_response"

    val all = List(request, response)
  }

  object ActionTopic {
    val request            = "action_request"
    val requestUnprocessed = "action_request_unprocessed"
    val response           = "action_response"

    val all = List(request, response)
  }

  object SagaTopic {
    val request          = "saga_request"
    val response         = "saga_response"
    val responseTopicMap = "saga_response_topic_map"
    val state            = "saga_state"
    val stateTransition  = "saga_state_transition"

    val all = List(request, response, state, stateTransition)
  }

  case class X[A]()
}

File: modules/shared/src/main/java/io/simplesource/shared/topics/TopicTypes.java
package io.simplesource.shared.topics;

import com.google.common.collect.Lists;

import java.util.List;

class TopicTypes {
    static class CommandTopic {
        String request  = "command_request";
        String response = "command_response";

        List<String> all = Lists.newArrayList(request, response);
    }

    static class ActionTopic {
        String request            = "action_request";
        String requestUnprocessed = "action_request_unprocessed";
        String response           = "action_response";
        List<String> all = Lists.newArrayList(request, response);
    }

    static class SagaTopic {
        String request          = "saga_request";
        String response         = "saga_response";
        String responseTopicMap = "saga_response_topic_map";
        String state            = "saga_state";
        String stateTransition  = "saga_state_transition";

        List<String> all = Lists.newArrayList(request, response, responseTopicMap, state, stateTransition);
        List<String> client = Lists.newArrayList(request, response);
    }
}
